tests/test_types.py

```diff
@@ -5370,11 +5371,39 @@ class Model(BaseModel):
 
 
 def test_annotated_default_value() -> None:
-    t = TypeAdapter(Annotated[List[str], Field(default=['foo', 'bar'])])
+    t = TypeAdapter(Annotated[List[int], Field(default=['1', '2'])])
 
-    # TODO: check the default value somehow with `validate_python`
-    # There is currently no way to do this
-    # See https://github.com/pydantic/pydantic-core/pull/643
+    r = t.get_default_value()
+    assert r is not None
+    assert r.value == ['1', '2']
 
     # insert_assert(t.json_schema())
-    assert t.json_schema() == {'type': 'array', 'items': {'type': 'string'}, 'default': ['foo', 'bar']}
+    assert t.json_schema() == {'type': 'array', 'items': {'type': 'integer'}, 'default': ['1', '2']}
+
+
+def test_annotated_default_value_validate_default() -> None:
+    t = TypeAdapter(Annotated[List[int], Field(default=['1', '2'])], config=ConfigDict(validate_default=True))
+
+    r = t.get_default_value()
+    assert r is not None
+    assert r.value == [1, 2]
+
+    # insert_assert(t.json_schema())
```

Comment:
Do we need this assertion? As it's commented

Reply:
yeah we leave those in there, makes it easy to know how to regenerate em